src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+import os
+import pathlib
+import tempfile
+import typing
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: typing.Dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.patches: typing.List[unittest.mock.patch] = []
+        for target, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
+            patch = unittest.mock.patch(target, mocked_path)
+            self.patches.append(patch)
+            patch.start()
+
+        with (self.chroot / "etc/insights-client/insights-client.motd").open("w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    @property
+    def chroot(self) -> pathlib.Path:
+        return pathlib.Path(self._dir.name)
+
+    def __del__(self):
+        for patch in self.patches:
+            patch.stop()
+        self._dir.cleanup()
+
+
+@pytest.fixture(scope="function")
+def mock_fs():
+    fs = MockFS()
+    yield fs
+    del fs
+
+
+def test_present(mock_fs):
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # The file gets created/symlinked when .registered & .unregistered do not exist
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/insights-client/insights-client.motd").samefile(
+        mock_fs.chroot / "etc/motd.d/insights-client"
+    )
```

Comment:
```suggestion
    assert mock_fs.patches["insights_client.MOTD_SRC"].new.samefile(
        mock_fs.patches["insights_client.MOTD_FILE"].new
    )
```

It feels more natural to me to swap the two files. I mean, what I am verifying is the MOTD_FILE and I want to know whether it’s the same file as its template, the MOTD_SRC. Just as I’d write e.g. `motd_path == "/etc/motd.d/insights-client"` rather than `"/etc/motd.d/insights-client" == motd_path` or `actual_path == expected_path` rather than `expected_path == actual_path`. I know that some languages and frameworks write it the opposite way, but I think it’s not the case of Python.

```suggestion
    assert (mock_fs.chroot / "etc/motd.d/insights-client").samefile(
        mock_fs.chroot / "etc/insights-client/insights-client.motd"
    )
```

How would it look like if variables were used to make the assert shorter?

```suggestion
    motd_src = mock_fs.chroot / "etc/insights-client/insights-client.motd"
    motd_file = mock_fs.chroot / "etc/motd.d/insights-client"
    assert motd_src.samefile(motd_file)
```

Reply:
Sure. It also makes it consistent with the `.samefile` asserts in the /dev/null tests.